File: Web-Frontend/src/api/userApi.js
import API from "./axios";



export const getAllUsers = (params) => API.get("/users", { params });

export const getStaff = (params) => API.get("/users/staff", { params });

export const getUserById = (id) => API.get(`/users/${id}`);

export const createUser = (data) => API.post("/users", data);

export const updateUser = (id, data) => API.put(`/users/${id}`, data);

export const toggleUserStatus = (id) => API.patch(`/users/${id}/toggle-status`);

export const getUserStats = () => API.get("/users/stats");

export const updateMyProfile = (formData) =>
  API.put("/users/profile/me", formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });

export const changeMyPassword = (data) => API.put("/users/password/me", data);

File: Mobile-App/mint_mobile/lib/api/user_api.dart
import 'package:dio/dio.dart';
import 'dio_client.dart';

class UserApi {
  static Future<Response> getAllUsers() =>
      DioClient.instance.get('/users');

  static Future<Response> getStaff() =>
      DioClient.instance.get('/users/staff');

  static Future<Response> getUserStats() =>
      DioClient.instance.get('/users/stats');

  // ✅ NEW: fetch a single user by ID
  static Future<Response> getUserById(String id) =>
      DioClient.instance.get('/users/$id');

  static Future<Response> createUser(Map<String, dynamic> data) =>
      DioClient.instance.post('/users', data: data);

  static Future<Response> updateUser(String id, Map<String, dynamic> data) =>
      DioClient.instance.put('/users/$id', data: data);

  static Future<Response> toggleUserStatus(String id) =>
      DioClient.instance.patch('/users/$id/toggle-status');

  static Future<Response> updateMyProfile(FormData formData) =>
      DioClient.instance.put(
        '/users/profile/me',
        data: formData,
        options: Options(contentType: 'multipart/form-data'),
      );

  static Future<Response> changeMyPassword(String current, String newPw) =>
      DioClient.instance.put('/users/password/me', data: {
        'currentPassword': current,
        'newPassword': newPw,
      });
}